src/libraries/System.Data.Common/src/System/Data/DataTableReader.cs

```diff
@@ -737,7 +737,16 @@ internal static DataTable GetSchemaTableFromDataTable(DataTable table)
             DataColumn ColumnSize = new DataColumn(SchemaTableColumn.ColumnSize, typeof(int));
             DataColumn NumericPrecision = new DataColumn(SchemaTableColumn.NumericPrecision, typeof(short));
             DataColumn NumericScale = new DataColumn(SchemaTableColumn.NumericScale, typeof(short));
-            DataColumn DataType = new DataColumn(SchemaTableColumn.DataType, typeof(Type));
+            DataColumn DataType = GetSystemTypeDataColumn();
+
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2111:RequiresDynamicallyAccessedMembers",
+                Justification = "The problem is Type.TypeInitializer which requires constructors on the Type instance." +
+                    "In this case the Type instance is about System.Type itself, so adding an explicit dependency on the same" +
+                    "annotation here.")]
+            [DynamicDependency(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors, typeof(Type))]
```

Comment:
Does the trimmer treat `PublicConstructors | NonPublicConstructors` as referencing static cctors?

I'm wondering if this `DynamicDependency` is truly necessary.